### DISCUSSION TITLE
高频的偶发异常 ThreadAbortException: Exception_WasThrown

### DISCUSSION THREAD

**Discussion by CharlesTaino1** [dotnetcore/FreeSql#1132] [General]
`List<Account> accountsList = DB.mysql_myproject.Select<Account>().Where(a => a.Status== 1).OrderBy(a => a.ID).ToList();`

System.TypeInitializationException
  HResult=0x80131534
  Message=The type initializer for 'xx.Entities.DB' threw an exception.

内部异常 1:
ThreadAbortException: Exception_WasThrown

**Comment by CharlesTaino1** [comment]
补充：偶尔可以正常获取数据，偶尔会报这个异常

**Reply by luoyunchong** [reply]
InnerException看下

**Reply by CharlesTaino1** [reply]
![image](https://user-images.githubusercontent.com/69567244/170920424-7bac438c-37c6-4603-a7a6-cf38498e501a.png)

**Comment by 2881099** [comment]
System.TypeInitializationException
HResult=0x80131534
Message=The type initializer for 'xx.Entities.DB' threw an exception.

这个跟初始化有关，描述环境，数据库版本、本地还是远程、.net runtime

**Reply by CharlesTaino1** [reply]
你好
本地：windows10 家庭版、.net 6.0、数据库Mysql8.0.22
测试服务器：windows10 专业版、.net 6.0、数据库Mysql8.0.22

程序在本地环境和测试环境，都会出现这样的问题。但是过一段时间，在不改任何代码的情况下又正常了。

**Reply by 2881099** [reply]
可能和你的代码结构有关，TypeInitializationException 一般指静态化就出错了，比如 static string xx = config.Get(“xx”) 

config.Get(“.xx”) 出错就回报 TypeInitializationException

**Reply by CharlesTaino1** [reply]
关键这个偶发很奇怪，就是这10分钟之内，不关重新运行多少次，都会有这个异常 ，过了一段时间，再启动可能又一切正常。

**Comment by CharlesTaino1** [comment]
这个问题仍然存在。。

**Reply by 2881099** [reply]
用新建项目测试，再不行可以打包上传
